lib/bin/build_env.sh

```diff
@@ -3,48 +3,48 @@
 # Detect the platform (only GitHub Actions in this case)
 if [ -n "$GITHUB_ACTIONS" ]; then
   platform=github_actions
-  branch=${SELECTIVE_BRANCH:-${GITHUB_HEAD_REF:-$GITHUB_REF_NAME}}
-  target_branch=${SELECTIVE_TARGET_BRANCH:-$GITHUB_BASE_REF}
+  branch=${GITHUB_HEAD_REF:-$GITHUB_REF_NAME}
+  target_branch=$GITHUB_BASE_REF
   actor=$GITHUB_ACTOR
-  sha=${SELECTIVE_SHA:-$GITHUB_SHA}
-  run_id=${SELECTIVE_RUN_ID:-$GITHUB_RUN_ID}
-  run_attempt=${SELECTIVE_RUN_ATTEMPT:-$GITHUB_RUN_ATTEMPT}
+  sha=$GITHUB_SHA
+  run_id=$GITHUB_RUN_ID
+  run_attempt=$GITHUB_RUN_ATTEMPT
   runner_id=$SELECTIVE_RUNNER_ID
 elif [ -n "$CIRCLECI" ]; then
   platform=circleci
-  branch=${SELECTIVE_BRANCH:-$CIRCLE_BRANCH}
-  target_branch=$SELECTIVE_TARGET_BRANCH
-  actor=${SELECTIVE_ACTOR:-${CIRCLE_USERNAME:-$CIRCLE_PR_USERNAME}}
-  sha=${SELECTIVE_SHA:-$CIRCLE_SHA1}
-  run_id=$SELECTIVE_RUN_ID
-  run_attempt=${SELECTIVE_RUN_ATTEMPT:-$CIRCLE_BUILD_NUM}
-  runner_id=${SELECTIVE_RUNNER_ID:-$CIRCLE_NODE_INDEX}
-else
-  platform=$SELECTIVE_PLATFORM
-  branch=$SELECTIVE_BRANCH
-  target_branch=$SELECTIVE_TARGET_BRANCH
-  actor=$SELECTIVE_ACTOR
-  sha=$SELECTIVE_SHA
-  run_id=$SELECTIVE_RUN_ID
-  run_attempt=$SELECTIVE_RUN_ATTEMPT
-  runner_id=$SELECTIVE_RUNNER_ID
+  branch=$CIRCLE_BRANCH
+  actor=${CIRCLE_USERNAME:-$CIRCLE_PR_USERNAME}
+  sha=$CIRCLE_SHA1
+  run_attempt=$CIRCLE_BUILD_NUM
+  runner_id=$CIRCLE_NODE_INDEX
+elif [ -n "$SEMAPHORE" ]; then
+  platform=semaphore
+  branch=${SEMAPHORE_GIT_PR_BRANCH:-$SEMAPHORE_GIT_BRANCH}
```

Comment:
Semaphore is weird

> In builds triggered by a Pull Request, the value of the SEMAPHORE_GIT_BRANCH is the name of the git branch targeted by the Pull Request.

https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_branch